keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +163,166 @@ Running this set of tests is optional for a per-provider basis. We would
 recommend it for providing extra confidence but the framework for
 upgrade/downgrade is provider agnostic.
 
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: CSIMigration, CSIMigration{cloud-provider}
+    - Components depending on the feature gate: kubelet, kube-controller-manager
+
+* **Does enabling the feature change any default behavior?**
+  Yes and No. If only CSIMigration feature flag is enabled, nothing will change on the cluster behavior. However, if CSIMigration && CSIMigration{cloud-provider} are both enabled, the behavior will change. The in-tree volume plugin that the cloud-provider use will be redirect to use the corresponding CSI driver. But from a user perspective, nothing will be noticed.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes. Once there is an error or something unexpected, users can disable the CSI migration feature flags to roll it back.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+The CSI migration will continue to function.
+
+* **Are there any tests for feature enablement/disablement?**
+We have CSI Migration e2e test for each plugin that are implemented and maintained by each driver maintainer. For core K8s, we have unit tests including but not limited to:
+   - `pkg/volume/csimigration/plugin_manager_test.go`
+   - All unit tests in the csi-translation-lib `staging/src/k8s.io/csi-translation-lib/translate_test.go`
+   - Controller test with Migration on CSI sidecars: external-provisioner, external-resizer
+     - provisioner: pkg/controller/controller_test.go#TestProvisionWithMigration
+     - resizer: pkg/resizer/csi_resizer_test.go#TestResizeMigratedPV
```

Comment:
All of those are checking "what if FG enabled" or "what if FG disabled".

Any transition tests? Would they bring additional value (depending on the question about storing state above).

Reply:
I am guessing you mean the upgrade/downgrade/version skews test? These tests are a requirement for the GA criteria.

Reply:
Note that we have some upgrade tests for storage here: https://github.com/kubernetes/kubernetes/tree/master/test/e2e/upgrades/storage

These will test a PVC created before migration enabled continues to function after upgrade. We could enhance this to add more feature coverage if needed.